docs/modules/ROOT/pages/accounts.adoc

```diff
@@ -1,632 +1,132 @@
-
-[,python]
-----
-await signer.send_transaction(account, registry.contract_address, 'set_L1_address', [NEW_ADDRESS])
-----
+Unlike Ethereum where accounts are derived from a private key, all Starknet accounts are contracts. This means there's no Externally Owned Account (EOA)
+concept on Starknet.
 
-NOTE: You can read more about how messages are structured and hashed in the https://github.com/OpenZeppelin/cairo-contracts/discussions/24[Account message scheme  discussion].
-For more information on the design choices and implementation of multicall, you can read the https://github.com/OpenZeppelin/cairo-contracts/discussions/27[How should Account multicall work discussion].
+Instead, the network features native account abstraction and signature validation happens at the contract level.
 
-The `\\__validate__` and `\\__execute__` methods accept the same arguments; however, `\\__execute__` returns a transaction response:
-
-[,cairo]
-----
-func __validate__(
-    call_array_len: felt, call_array: AccountCallArray*, calldata_len: felt, calldata: felt*) {
-}
-
-func __execute__(
-    call_array_len: felt, call_array: AccountCallArray*, calldata_len: felt, calldata: felt*
-) -> (response_len: felt, response: felt*) {
-}
-----
+For a general overview of the account abstraction, see
+https://docs.starknet.io/documentation/architecture_and_concepts/Accounts/introduction/[Starknet's documentation].
+A more detailed discussion on the topic can be found in
+https://community.starknet.io/t/starknet-account-abstraction-model-part-1/781[Starknet Shaman's forum].
 
-Where:
+TIP: For detailed information on the usage and implementation check the xref:/api/account.adoc[API Reference] section.
 
-* `call_array_len` is the number of calls.
-* `call_array` is an array representing each `Call`.
-* `calldata_len` is the number of calldata parameters.
-* `calldata` is an array representing the function parameters.
+== Basic Account
 
-NOTE: The scheme of building multicall transactions within the `\\__execute__` method will change once StarkNet allows for pointers in struct arrays.
-In which case, multiple transactions can be passed to (as opposed to built within) `\\__execute__`.
+As we mentioned, accounts in Starknet are smart contracts, and so they can be deployed and interacted
+with like any other contract, and can be extended to implement any custom logic. However, accounts are a special type
+of contract that are used to validate and execute transactions, for which they must implement a minimum set of functions.
 
-There's a fourth canonical entrypoint for accounts, the `\\__validate_deploy__` method. It is **only callable by the protocol** during the execution of a `DeployAccount` type of transaction, but not by any other contract. This entrypoint is for counterfactual deployments.
+=== Account methods
 
-=== Counterfactual Deployments
+The first two following methods are protocol-level enforced functions that must be implemented by any account contract.
```

Comment:
> Or even dropping the intro paragraph and just include "protocol-enforced" in the function description

i think this might not be very clear on what it means. what about:

```suggestion
SRC6 standard account contracts must implement the following methods, the first two being enforced by the protocol.
```
or similar. it also addresses the discussed difference between "protocol minimal viable accounts" and standard ones

Reply:
> it also addresses the discussed difference between "protocol minimal viable accounts" and standard ones

Yes, much better!